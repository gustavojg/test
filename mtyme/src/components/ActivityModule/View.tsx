import React from "react";
import styles from "@/styles/ActivityView.module.css";
export type ActivityViewProps = {
  type: string;
  currenTime: string;
  periodTime: string;
  periodText: string;
  bgColor: string;
  icon: string;
};

export const ActivityView: React.FC<ActivityViewProps> = (props) => {
  const { type, currenTime, periodTime, periodText, icon, bgColor } = props;
  return (
    <div className={styles.mainContainer}>
      <div
        className={styles.header}
        style={{
          backgroundColor: bgColor,
        }}
      >
        <img className={styles.icon} src={icon} alt={type} />
      </div>
      <div className={styles.infoContainer}>
        <div className={styles.headerInfo}>
          <div className={styles.typetext}>{type}</div>
          <div className={styles.dotMenu}>
            <img src="/images/icon-ellipsis.svg" alt="menu" />
          </div>
        </div>
        <div className={styles.hoursText}>{currenTime}hrs</div>
        <div className={styles.periodText}>
          {periodText} - {periodTime}hrs
        </div>
      </div>
    </div>
  );
};
