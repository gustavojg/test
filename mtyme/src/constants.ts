export const DAILY: string = "DAILY";
export const WEEKLY: string = "WEEKLY";
export const MONTHLY: string = "MONTHLY";

export const TIME_TYPES = {
  WORK: "WORK",
  PLAY: "PLAY",
  STUDY: "STUDY",
  EXERCISE: "EXERCISE",
  SOCIAL: "SOCIAL",
  SELF_CARE: "SELF_CARE",
};
