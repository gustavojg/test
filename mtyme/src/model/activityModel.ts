export type Activity = {
  type: string;
  currentDuration: number;
  perriodDuration: number;
  periodType: string;
};
