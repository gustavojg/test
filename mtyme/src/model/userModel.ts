export type User = {
    id: number;
    name: string;
    lastName: string;
    image: string;
}