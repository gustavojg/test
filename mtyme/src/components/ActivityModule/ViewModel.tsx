import { getPeriodText, getStaticInfo } from "@/lib/staticInfo";
import { Activity } from "@/model/activityModel";

export const ActivityViewModel = ({
  currentDuration,
  perriodDuration,
  periodType,
  type,
}: Activity) => {
  const { title, icon, color } = getStaticInfo(type);
  return {
    type: title,
    currenTime: currentDuration.toString(),
    periodTime: perriodDuration.toString(),
    periodText: getPeriodText(periodType),
    icon,
    bgColor: color,
  };
};
