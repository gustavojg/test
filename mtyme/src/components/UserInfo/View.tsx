import { User } from "@/model/userModel";
import React from "react";
import styles from "@/styles/UserInfo.module.css";
import { DAILY, MONTHLY, WEEKLY } from "@/constants";
import Image from "next/image";

type UserViewProps = {
  user: User | null;
  periodType: string;
  onPeriodSelect: (p: string) => void;
};

export const UserView: React.FC<UserViewProps> = ({
  user,
  onPeriodSelect,
  periodType,
}) => {
  if (!user) {
    return <div>Loading...</div>;
  }
  return (
    <div className={styles.mainContainer}>
      <div className={styles.container}>
        <Image
          className={styles.userImage}
          src={user.image}
          alt={user.name}
          width={100}
          height={100}
          priority
        />
        <div className={styles.title}>Report for</div>
        <div className={styles.userName}>{user.name}</div>
        <div className={styles.userName}>{user.lastName}</div>
      </div>
      <div className={styles.buttonsContainer}>
        <a
          className={`${styles.periodButton} ${
            periodType === DAILY ? styles.selected : ""
          }`}
          onClick={() => onPeriodSelect(DAILY)}
        >
          Daily
        </a>
        <a
          className={`${styles.periodButton} ${
            periodType === WEEKLY ? styles.selected : ""
          }`}
          onClick={() => onPeriodSelect(WEEKLY)}
        >
          Weekly
        </a>
        <a
          className={`${styles.periodButton} ${
            periodType === MONTHLY ? styles.selected : ""
          }`}
          onClick={() => onPeriodSelect(MONTHLY)}
        >
          Monthly
        </a>
      </div>
    </div>
  );
};
