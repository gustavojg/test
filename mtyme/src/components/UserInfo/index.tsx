import React from "react";
import { UserView } from "./View";
import { User } from "@/model/userModel";

interface UserInfoProps {
  user: User | null;
  periodType: string;
  onPeriodSelect: (period: string) => void;
}

export const UserInfo: React.FC<UserInfoProps> = (props) => {
  return <UserView {...props} />;
};
