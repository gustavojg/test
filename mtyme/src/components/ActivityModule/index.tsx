import React from "react";
import { ActivityView } from "./View";
import { ActivityViewModel } from "./ViewModel";
import { Activity } from "@/model/activityModel";

export const ActivityInfo: React.FC<Activity> = (props) => {
  const activityViewModel = ActivityViewModel(props);
  return <ActivityView {...activityViewModel} />;
};
