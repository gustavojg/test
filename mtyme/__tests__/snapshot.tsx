/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import Home from "@/pages/home/index";
import { UserInfo, UserInfoProps } from "@/components/UserInfo";
import { DAILY, TIME_TYPES } from "@/constants";
import { ActivityInfo } from "@/components/ActivityModule";
import { Activity } from "@/model/activityModel";

const activity: Activity = {
  type: TIME_TYPES.WORK,
  currentDuration: 10,
  perriodDuration: 20,
  periodType: DAILY,
};

const userInfoProps: UserInfoProps = {
  user: {id: 1, name: "John", lastName: "Doe", image: "/images/users/image-jeremy.png"},
  periodType: DAILY,
  onPeriodSelect: jest.fn()
};

it("renders homepage unchanged", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});


it("renders UserInfo component", () => {
  const { container } = render(<UserInfo {...userInfoProps} />);
  expect(container).toMatchSnapshot();
});

it("renders ActivityInfo component", () => {
  const { container } = render(<ActivityInfo {...activity} />);
  expect(container).toMatchSnapshot();
});
