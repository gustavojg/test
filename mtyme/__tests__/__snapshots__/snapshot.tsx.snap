// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ActivityInfo component 1`] = `
<div>
  <div
    class="mainContainer"
  >
    <div
      class="header"
      style="background-color: rgb(102, 102, 102);"
    >
      <img
        alt="Work"
        class="icon"
        src="/images/icon-work.svg"
      />
    </div>
    <div
      class="infoContainer"
    >
      <div
        class="headerInfo"
      >
        <div
          class="typetext"
        >
          Work
        </div>
        <div
          class="dotMenu"
        >
          <img
            alt="menu"
            src="/images/icon-ellipsis.svg"
          />
        </div>
      </div>
      <div
        class="hoursText"
      >
        10
        hrs
      </div>
      <div
        class="periodText"
      >
        Yesterday
         - 
        20
        hrs
      </div>
    </div>
  </div>
</div>
`;

exports[`renders UserInfo component 1`] = `
<div>
  <div
    class="mainContainer"
  >
    <div
      class="container"
    >
      <img
        alt="John"
        class="userImage"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="100"
        src="/_next/image?url=%2Fimages%2Fusers%2Fimage-jeremy.png&w=256&q=75"
        srcset="/_next/image?url=%2Fimages%2Fusers%2Fimage-jeremy.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Fusers%2Fimage-jeremy.png&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
      <div
        class="title"
      >
        Report for
      </div>
      <div
        class="userName"
      >
        John
      </div>
      <div
        class="userName"
      >
        Doe
      </div>
    </div>
    <div
      class="buttonsContainer"
    >
      <a
        class="periodButton selected"
      >
        Daily
      </a>
      <a
        class="periodButton "
      >
        Weekly
      </a>
      <a
        class="periodButton "
      >
        Monthly
      </a>
    </div>
  </div>
</div>
`;

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="main className"
  >
    <div
      class="activityContainer gridContainer"
    >
      <div>
        Loading...
      </div>
    </div>
  </main>
</div>
`;
